### PR TITLE
feat:  nydusify can dump metrics created by converter

### PR DESCRIPTION
**Pull request by Desiki-high**
### We are introducing a new arg called "output-json" that will allow us to save the metrics generated by convert. This will enable us to successfully implement https://github.com/dragonflyoss/image-service/issues/1263 through nydusify. 
For additional information, please see https://github.com/goharbor/acceleration-service/pull/133.

### CODE REVIEW

**Comment by Desiki-high** [comment]
The output json like this:
image

File: contrib/nydusify/cmd/nydusify.go [hunk 1]
```diff
@@ -409,6 +409,12 @@ func main() {
 					Usage:   "Path to the nydus-image binary, default to search in PATH",
 					EnvVars: []string{"NYDUS_IMAGE"},
 				},
+				&cli.BoolFlag{
+					Name:    "output-json",
+					Value:   false,
+					Usage:   "Dump the metric colledted in convert to local json file: output.json",
```
[the remark this answers is not in the document]
Comment (imeoer) [reply contrib/nydusify/cmd/nydusify.go:415]: Enable saving the metrics collected during conversion in JSON file output.json.

[the remark this answers is not in the document]
Comment (Desiki-high) [reply contrib/nydusify/cmd/nydusify.go]: fixed

File: contrib/nydusify/pkg/converter/metric.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+	"encoding/json"
+	"os"
+
+	"github.com/goharbor/acceleration-service/pkg/converter"
+	"github.com/pkg/errors"
+)
+
+func dumpMetric(metric *converter.Metric) error {
+
+	file, err := os.Create("output.json")
```
[the remark this answers is not in the document]
Comment (Desiki-high) [reply contrib/nydusify/pkg/converter/metric.go]: fixed

File: contrib/nydusify/pkg/converter/metric.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+		return errors.Wrap(err, "Error creating file for metric")
+	}
+	defer file.Close()
+
+	encoder := json.NewEncoder(file)
+	err = encoder.Encode(metric)
+	if err != nil {
+		return errors.Wrap(err, "Error encoding JSON from metric")
+	}
+	return nil
```
[the remark this answers is not in the document]
Comment (Desiki-high) [reply contrib/nydusify/pkg/converter/metric.go:27]: fixed

**Review by imeoer — APPROVED** [review]

**Pull request merged**